src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java

```diff
@@ -330,42 +329,72 @@ public Flow updateFlow(FlowOperationsCarrier carrier, FlowDto flow) throws FlowN
-                && !flowDto.getPathComputationStrategy().equals(flow.getPathComputationStrategy());
-        boolean changedMaxLatency = flowDto.getMaxLatency() != null
-                && !flowDto.getMaxLatency().equals(flow.getMaxLatency());
-        boolean strategyIsMaxLatency = PathComputationStrategy.MAX_LATENCY.equals(flowDto.getPathComputationStrategy())
-                || flowDto.getPathComputationStrategy() == null
+    UpdateFlowResult.UpdateFlowResultBuilder prepareFlowUpdateResult(FlowPatch flowPatch, Flow flow) {
+        boolean changedStrategy = flowPatch.getPathComputationStrategy() != null
+                && !flowPatch.getPathComputationStrategy().equals(flow.getPathComputationStrategy());
+        boolean changedMaxLatency = flowPatch.getMaxLatency() != null
+                && !flowPatch.getMaxLatency().equals(flow.getMaxLatency());
+        boolean strategyIsMaxLatency =
+                PathComputationStrategy.MAX_LATENCY.equals(flowPatch.getPathComputationStrategy())
+                || flowPatch.getPathComputationStrategy() == null
                 && PathComputationStrategy.MAX_LATENCY.equals(flow.getPathComputationStrategy());
 
-        String reason = null;
-        if (changedStrategy) {
-            reason = format("initiated via Northbound, path computation strategy was changed from %s to %s",
-                    flow.getPathComputationStrategy(), flowDto.getPathComputationStrategy());
-        } else if (changedMaxLatency && strategyIsMaxLatency) {
-            reason = format("initiated via Northbound, max latency was changed from %d to %d",
-                    flow.getMaxLatency(), flowDto.getMaxLatency());
-        }
+        boolean changedSourceSwitch = flowPatch.getSourceSwitch() != null
+                && !flow.getSrcSwitch().getSwitchId().equals(flowPatch.getSourceSwitch());
+        boolean changedSourcePort = flowPatch.getSourcePort() != null
+                && flow.getSrcPort() != flowPatch.getSourcePort();
+        boolean changedSourceVlan = flowPatch.getSourceVlan() != null
+                && flow.getSrcVlan() != flowPatch.getSourceVlan();
+        boolean changedSourceEndpoint = changedSourceSwitch || changedSourcePort || changedSourceVlan;
+
+        boolean changedDestinationSwitch = flowPatch.getDestinationSwitch() != null
+                && !flow.getDestSwitch().getSwitchId().equals(flowPatch.getDestinationSwitch());
+        boolean changedDestinationPort = flowPatch.getDestinationPort() != null
+                && flow.getDestPort() != flowPatch.getDestinationPort();
+        boolean changedDestinationVlan = flowPatch.getDestinationVlan() != null
+                && flow.getDestVlan() != flowPatch.getDestinationVlan();
+        boolean changedDestinationEndpoint
+                = changedDestinationSwitch || changedDestinationPort || changedDestinationVlan;
+
+        boolean addedDiverseFlowId = flowPatch.getDiverseFlowId() != null;
```

Comment:
what about:
1. comparing with current DiverseFlowId?
2. setting DiverseFlowId = null to do not diverse with any flow?

Reply:
Fixed